tools/slo-generator/slo_generator/backends/stackdriver_service_monitoring.py

```diff
@@ -0,0 +1,663 @@
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#            http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+`stackdriver_service_monitoring.py`
+Stackdriver Service Monitoring exporter class.
+"""
+import difflib
+import json
+import logging
+import os
+
+import google.api_core.exceptions
+from google.cloud.monitoring_v3 import ServiceMonitoringServiceClient
+from google.protobuf.json_format import MessageToJson
+
+from slo_generator.backends.stackdriver import StackdriverBackend
+from slo_generator.utils import dict_snake_to_caml
+
+LOGGER = logging.getLogger(__name__)
+
+SID_GAE = 'gae:{project_id}_{module_id}'
+SID_CLOUD_ENDPOINT = 'ist:{project_id}-{service}'
+SID_CLUSTER_ISTIO = (
+    'ist:{project_id}-zone-{location}-{cluster_name}-{service_namespace}-'
+    '{service_name}')
+SID_MESH_ISTIO = ('ist:{mesh_uid}-{service_namespace}-{service_name}')
+
+
+class StackdriverServiceMonitoringBackend:
```

Comment:
What's the relationship with StackdriverBackend?

Reply:
None, this class queries the Stackdriver Monitoring API, the other queries the Stackdriver Service Monitoring API. See `samples/stackdriver_service_monitoring` and `samples/stackdriver_monitoring`.